tests/PHPStan/Rules/Arrays/NonexistentOffsetInArrayDimFetchRuleTest.php

```diff
@@ -111,10 +111,6 @@ public function testRule(): void
 				'Cannot access offset \'a\' on Closure(): void.',
 				253,
 			],
-			[
-				'Cannot access offset \'a\' on array{a: 1, b: 1}|(Closure(): void).',
-				258,
-			],
```

Comment:
Calling closure or object (without ArrayAccess) causes a fatal error, so this case shouldn't be ignored.
https://3v4l.org/lFcCq

I think it might be better to have another condition to the type, whether isOffestAccessible->no throws an fatal error or not.
This might lead to improvements in issues like https://github.com/phpstan/phpstan/issues/8388

Reply:
It's better to open a new issue about this, comments on old merged PRs will get lost :)

Reply:
Thank you! Done https://github.com/phpstan/phpstan/issues/8393